app/services/xapi/completed_product_service.rb

```diff
@@ -0,0 +1,29 @@
+module Xapi
+  class CompletedProductService < Xapi::BaseService
```

Comment:
Add `attr_readers` for `@agent`, `@object`, `@statement` to facilitate testing and debugging.

Reply:
I'm not sure about this - but I do not know how we plan to test things like these. @houdyerpierre maybe you could add some tests to see if the code is testable and to visualise your idea how to test it?